cads_adaptors/adaptors/cadsobs.py

```diff
@@ -0,0 +1,58 @@
+from pathlib import Path
+
+from cads_adaptors.adaptors.cds import AbstractCdsAdaptor
+from cads_adaptors.mapping import apply_mapping
+
+
+class ObservationsAdaptor(AbstractCdsAdaptor):
+    def retrieve(self, request):
+        from cdsobs.retrieve.api import retrieve_observations
+        from cdsobs.retrieve.models import RetrieveArgs
+        from cdsobs.utils.utils import get_database_session
+
+        # Maps observation_type to source
+        mapped_request = apply_mapping(request, self.mapping)
```

Comment:
Could you merge from main and then change this to:
```suggestion
        self._pre_retrieve(request=request)
```